src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Filter } from '../../enums/Filter';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  filter: Filter,
+  setFilter: (value: Filter) => void;
+  onClear: () => void;
+  todos: Todo[];
+};
+
+export const TodoFilter: React.FC<Props> = React.memo(
+  ({
+    filter, setFilter, onClear, todos,
+  }) => {
+    const isAnyTodoCompleted = todos.some(todo => todo.completed);
```

Comment:
it's best practice)
pls, use it in all cases
```suggestion (props) => {
    const {
     filter,
     setFilter,
     onClear,
     todos,
  } = props
    const isAnyTodoCompleted = todos.some(todo => todo.completed);
```